05_iigo/index.tex

```diff
@@ -500,4 +500,226 @@ \subsection{Legislative Branch}
+        x \\
+        y
+    \end{bmatrix}   
+    >= 0  
+\]
+The addition of the "at least" adds a little bit of complexity to our matrix schema. In that we may not be able to simply check for equality to a constant, and may instead need to check for a certain condition.
+To make these comparisons easier, we have decided to ensure that whatever condition, equality or inequality, we would compare to 0.
+However if our tax rule looked like this (for whatever reason):
+\linebreak
+\linebreak
+"If you are expected to pay x amount of tax, the amount of tax you pay must be at least x + 5"
+\linebreak
+\linebreak
+Then it would be impossible to capture that using the current matrix calculation, so to capture these cases we introduce a constant to the input list, as below.
+\[
+    \begin{bmatrix}
+        -1 & 1 & -5
+    \end{bmatrix}
+    \begin{bmatrix}
+        x \\
+        y \\
+        1
+    \end{bmatrix}   
+    >= 0  
+\]
+We have now managed to capture any linear equation or inequality with any constant shift. This is where we decided to limit the scope of our rules from a technical aspect.
+The final section of our schema is how to inform an agent whether it should be looking for equality or inequality.
+For this, we package with the rule-matrix an "auxiliary" vector containing a list of comparison codes.
+\begin{table}[h]
+    \centering
+    \begin{tabular}{ll}
+    \hline
+    \multicolumn{1}{|l}{Aux code} & \multicolumn{1}{l|}{Meaning} \\ \hline
+    0                             & = 0                          \\
+    1                             & \textgreater 0               \\
+    2                             & \textgreater{}= 0            \\
+    3                             & != 0                         \\
+    4                             & real                        
+    \end{tabular}
+\end{table}
+\linebreak
+The aux code 4 is reserved for when we want the raw result of the matrix multiplication, which is used sparingly for certain rules.
+We have now arrived at the representation of rules we are using for our project. For the rule above:
+\[
+    \textrm{Rule matrix: }
+    \begin{bmatrix}
+        -1 & 1 & -5
+    \end{bmatrix}
+    \textrm{   Variables to fill:  }
+    \begin{bmatrix}
+        \textrm{Amount of tax agent is asked to pay} \\
+        \textrm{Amount of tax the agent is actually paying} \\
+        1
+    \end{bmatrix}     
+\]
+\[
+    \textrm{   Auxiliary vector: } 
+    \begin{bmatrix}
+        2 
+    \end{bmatrix}   
+\]
+Let us revist our tax rule once more, adding a further layer of complexity.
+\linebreak
+\linebreak
+"You must pay in tax at least half as many resources as you have and if you've been told to pay tax x you must pay at least tax of x."
+In this example there are two separate conditions. We can write both conditions as separate rules.
+Condition 1:
+\[
+    \textrm{Rule matrix: }
+    \begin{bmatrix}
+        0 & 2 & -1 & 0
+    \end{bmatrix}
+    \textrm{   Variables to fill:  }
```

Comment:
plz plz go to another equation :) 
do NOT write on the same line

`\begin{equation} \end{equation}` plzzzzz